x/blob/client/cli/payforblob.go

```diff
@@ -28,69 +29,149 @@ const (
 	// FlagNamespaceVersion allows the user to override the namespace version when
 	// submitting a PayForBlob.
 	FlagNamespaceVersion = "namespace-version"
+
+	// FlagFileInput allows the user to provide file path to the json file
+	// for submitting multiple blobs.
+	FlagFileInput = "input-file"
 )
 
 func CmdPayForBlob() *cobra.Command {
 	cmd := &cobra.Command{
-		Use: "PayForBlobs namespaceID blob",
+		Use: "PayForBlobs [namespaceID blob]",
 		// This example command can be run in a new terminal after running single-node.sh
 		Example: "celestia-appd tx blob PayForBlobs 0x00010203040506070809 0x48656c6c6f2c20576f726c6421 \\\n" +
 			"\t--chain-id private \\\n" +
 			"\t--from validator \\\n" +
 			"\t--keyring-backend test \\\n" +
 			"\t--fees 21000utia \\\n" +
-			"\t--yes",
-		Short: "Pay for a data blob to be published to Celestia.",
-		Long: "Pay for a data blob to be published to Celestia.\n" +
-			"namespaceID is the user-specifiable portion of a version 0 namespace. It must be a hex encoded string of 10 bytes.\n" +
-			"blob must be a hex encoded string of any length.\n" +
-			// TODO: allow for more than one blob to be sumbmitted via the CLI
-			"This command currently only supports a single blob per invocation.\n",
+			"\t--yes \\\n" +
+			"\n" +
+			"celestia-appd tx blob PayForBlobs --input-file path/to/blobs.json \\\n" +
+			"\t--chain-id private \\\n" +
+			"\t--from validator \\\n" +
+			"\t--keyring-backend test \\\n" +
+			"\t--fees 21000utia \\\n" +
+			"\t--yes \\\n" +
+			"\\\n",
+		Short: "Pay for a data blob(s) to be published to Celestia.",
+		Long: "Pay for a data blob(s) to be published to Celestia.\n" +
+			"User can use namespaceID and blob as argument for single blob submission \n" +
+			"or use --input-file flag with the path to a json file for multiple blobs submission, \n" +
+			`where the json file contains: 
+
+		{
+			"Blobs": [
+				{
+					"namespaceId": "0x00010203040506070809",
+					"blob": "0x48656c6c6f2c20576f726c6421"
+				},
+				{
+					"namespaceId": "0x00010203040506070809",
+					"blob": "0x48656c6c6f2c20576f726c6421"
+				}
+			]
+		}
+
+		namespaceID is the user-specifiable portion of a version 0 namespace. It must be a hex encoded string of 10 bytes.\n
+		blob must be a hex encoded string of any length.\n
+		`,
 		Aliases: []string{"PayForBlob"},
 		Args: func(cmd *cobra.Command, args []string) error {
+			path, err := cmd.Flags().GetString(FlagFileInput)
+			if err != nil {
+				return err
+			}
+
+			// If there is a file path input we'll check for the file extension
+			if path != "" {
+				if filepath.Ext(path) != ".json" {
+					return fmt.Errorf("invalid file extension, require json got %s", filepath.Ext(path))
+				}
+
+				return nil
+			}
+
 			if len(args) < 2 {
-				return fmt.Errorf("PayForBlobs requires two arguments: namespaceID and blob")
+				return fmt.Errorf("PayForBlobs requires two argument: namespaceID and blob")
```

Comment:
if we're going to use fmt.Errorf, then we should include something to format in the string other wise I think the linter is going to block us

Reply:
updated as request